### PR TITLE
feat: add solana target chain example

### PR DESCRIPTION
**Pull request by keyvankhademi**
This adds a simple send usd example for solana target chain

### CODE REVIEW

**Comment by Reisen** [comment]
Failed to generate code suggestions for PR

**Comment by Reisen** [comment]
Failed to generate code suggestions for PR

**Comment by Reisen** [comment]
Failed to generate code suggestions for PR

**Comment by Reisen** [comment]
Failed to generate code suggestions for PR

**Comment by Reisen** [comment]
Failed to generate code suggestions for PR

**Comment by Reisen** [comment]
Failed to generate code suggestions for PR

**Comment by Reisen** [comment]
Failed to generate code suggestions for PR

File: target_chains/solana/examples/send_usd/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+[features]
+no-entrypoint = []
+no-idl = []
+no-log-ix-name = []
+cpi = ["no-entrypoint"]
+default = []
+
+[dependencies]
+anchor-lang = { workspace = true }
+ahash = "=0.8.6"
```
Comment (guibescos) [inline target_chains/solana/examples/send_usd/Cargo.toml:20]: I think we don't need this anymore

File: target_chains/solana/examples/send_usd/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+declare_id!("2e5gZD3suxgJgkCg4pkoogxDKszy1SAwokz8mNeZUj4M");
+
+pub const MAXIMUM_AGE: u64 = 24 * 3600 * 365;
+pub const FEED_ID: &str = "0xef0d8b6fda2ceba41da15d4095d1da392a0d2f8ed0c6c7bc0f4cfac8c280b56d";
+
+#[program]
+pub mod my_first_pyth_app {
+    use {
+        super::*,
+        pyth_solana_receiver_sdk::price_update::VerificationLevel,
```
Comment (guibescos) [inline target_chains/solana/examples/send_usd/src/lib.rs:24]: This import should probably be at the very top

File: target_chains/solana/examples/send_usd/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    };
+
+    pub fn send(ctx: Context<Send>, amount_in_usd: u64) -> Result<()> {
+        let price_update = &mut ctx.accounts.price_update;
+        let price = price_update.get_price_no_older_than_with_custom_verification_level(
```
Comment (guibescos) [inline target_chains/solana/examples/send_usd/src/lib.rs:29]: Use `get_price_no_older_than` in this default example

File: target_chains/solana/examples/send_usd/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+[package]
+name = "my_first_pyth_app"
```
Comment (guibescos) [inline target_chains/solana/examples/send_usd/Cargo.toml:2]: mmmm maybe rename this every to send_usd?

**Review by guibescos — COMMENTED** [review]
I think you add the path to this program in https://www.anchor-lang.com/docs/manifest#workspace so idl generation works with anchor build

File: target_chains/solana/examples/send_usd/Cargo.toml (hunk 1 above)
Comment (keyvankhademi) [reply target_chains/solana/examples/send_usd/Cargo.toml:20]: Fixed.

File: target_chains/solana/examples/send_usd/src/lib.rs (hunk 3 above)
Comment (keyvankhademi) [reply target_chains/solana/examples/send_usd/src/lib.rs:29]: fixed

File: target_chains/solana/examples/send_usd/Cargo.toml (hunk 4 above)
Comment (keyvankhademi) [reply target_chains/solana/examples/send_usd/Cargo.toml]: done

File: target_chains/solana/examples/send_usd/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    },
+    pyth_solana_receiver_sdk::price_update::{
+        get_feed_id_from_hex,
+        PriceUpdateV2,
+    },
+};
+
+declare_id!("2e5gZD3suxgJgkCg4pkoogxDKszy1SAwokz8mNeZUj4M");
+
+pub const MAXIMUM_AGE: u64 = 24 * 3600 * 365;
```
Comment (guibescos) [inline target_chains/solana/examples/send_usd/src/lib.rs:17]: Can we change this to 1 hour

File: target_chains/solana/examples/send_usd/Cargo.toml [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+no-entrypoint = []
+no-idl = []
+no-log-ix-name = []
+cpi = ["no-entrypoint"]
+default = []
+
+[dependencies]
+anchor-lang = { workspace = true }
+solana-program = { workspace = true }
+pyth-solana-receiver-sdk = { path = "../../pyth_solana_receiver_sdk"}
```
Comment (guibescos) [inline target_chains/solana/examples/send_usd/Cargo.toml:21]: I feel like this should use the public crate since that's how someone would use the sdk

**Review by guibescos — APPROVED** [review]

File: target_chains/solana/examples/send_usd/src/lib.rs (hunk 5 above)
Comment (keyvankhademi) [reply target_chains/solana/examples/send_usd/src/lib.rs:17]: Done.

**Pull request merged**